eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -424,6 +424,13 @@ jobs:
       continueOnError: true
       condition: succeededOrFailed()
 
+  - ${{ if or(ne(variables['System.TeamProject'], 'internal'), eq(variables['Build.Reason'], 'PullRequest')) }}:
+    - publish: $(Build.StagingDirectory)/BuildLogs
+      artifact: $(Agent.JobName)_BuildLogs_Attempt$(System.JobAttempt)
+      displayName: Publish BuildLogs
+      continueOnError: true
+      condition: succeededOrFailed()
```

Comment:
Wouldn't we want to have `always()` here? irrc the difference is that `succeededOrFailed()` won't trigger if the run is canceled, but maybe just trying to always publish is more straight-forward?

Reply:
Makes sense. I will put this to the artifact upload though. Leave only logs for cancelled builds.